codegen/arrai/svc_types.arrai

```diff
@@ -50,21 +50,35 @@ let orderedTypes = \types
                     cond {optParams: $`func (t *${typename}) UnmarshalJSON(data []byte) error {
                         inner := struct {
                             ${entity('attrDefs') where sysl.isPtr(.@value) orderby .@ >> \(@: key, @value: attrDef)
-                                $'${go.name(key)} ${go.type(attrDef)} `json:"${go.name(key)},omitempty"`'
+                                let jsonTag = 
+                                    cond attrDef {
+                                        {'attrs': {'json_tag': tag, ...}, ...}: tag('s').s,
+                                        _: key,
+                                    };
+                                $'${go.name(key)} ${go.type(attrDef)} `json:"${jsonTag},omitempty"`'
                             ::\i}
                             ${entity('attrDefs') where !sysl.isPtr(.@value) orderby .@ >> \(@: key, @value: attrDef)
-                                $'${go.name(key)} *${go.type(attrDef)} `json:"${go.name(key)},omitempty"`'
+                                let jsonTag = 
```

Comment:
These blocks are near identical. Can we simplify this to something like:
```
${entity('attrDefs') orderby .@ >> \(@: key, @value: attrDef)
    $'${go.name(key)} ${go.type(attrDef)} `json:"${go.name(key)},omitempty"`'
    let jsonTag = 
        cond attrDef {
            {'attrs': {'json_tag': tag, ...}, ...}: tag('s').s,
            _: key,
        };
    $'${go.name(key)} ${cond { !sysl.isPtr(.@value) : '*', _ : ''}}${go.type(attrDef)} `json:"${jsonTag},omitempty"`'
::\i}
```

Reply:
Extracted the duplicates into a new code block.